### PR TITLE
Added golden test suite

### PR DESCRIPTION
**Pull request by MegaIng**
Many of the tests in this base list are based on the once found in the test suite for the k-spec written by Grape.

### CODE REVIEW

**Review by EnderShadow — DISMISSED** [review]

**Review by EnderShadow — DISMISSED** [review]

**Comment by EnderShadow** [comment]
No, don't explicitly use 3.10 in the shebang. There's already a runtime check for the version. Hard coding 3.10 makes it break on future versions of python

**Comment by MegaIng** [comment]
@EnderShadow That is the only way to get it to work while directly executing. If you want to use a later version of python, explicitly call it with that version of python. We can also change this back to `python3` if ubuntu ever upgrades to `3.10` by default. If you don't agree with hardcoding `3.10`, I would suggest to just remove the shebang and always require a call to python.

File: etc-as.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/env python3.10
```
Comment (EnderShadow) [inline etc-as.py:1]:
```suggestion
#!/usr/bin/env python3
```

File: tests/golden/golden_tester_generic.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import subprocess
+import tempfile
+from pathlib import Path
+from pprint import pprint
+from dataclasses import dataclass
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-c", "--command", action="store", default="python3.10 etc-as.py",
```
Comment (EnderShadow) [inline tests/golden/golden_tester_generic.py:15]:
```suggestion
    parser.add_argument("-c", "--command", action="store", default="python3 etc-as.py",
```

File: tests/golden/golden_tester_generic.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3.10
```
Comment (EnderShadow) [inline tests/golden/golden_tester_generic.py:1]:
```suggestion
#!/usr/bin/env python3
```

**Review by EnderShadow — CHANGES_REQUESTED** [review]
We already have a version check for 3.10 in the code. Hard coding 3.10 breaks the code for future versions of python.

**Comment by EnderShadow** [comment]
@MegaIng By that same logic, we should leave the shebang and require explicit use of python3.10 to run it if you have multiple versions installed

**Comment by JKTKops** [comment]
Personally I'd prefer leaving the shebang with 3.10 (can always change it in the future if needed). I'm working on a system (via `make`, which is good for this kind of thing despite its flaws) which should get beginners up and running as fast as possible via

1. checking for python3.10 or better
2. walking through the install if not present
3. Copying a runner script to /usr/bin
4. copying the python stuff to /usr/lib or wherever
5. runner script (`etc-as`) discovers and uses latest python version explicitly

My biggest complaint with this method is only that it requires `sudo`. It is no longer a challenge to keep a beginner up to date and we don't have to worry about the beginner experience outside of the runner script which can handle everything.

**Comment by EnderShadow** [comment]
I no longer care enough about the whole 3 vs 3.10 thing

**Review by EnderShadow — DISMISSED** [review]

**Review by JKTKops — DISMISSED** [review]

**Review by EnderShadow — APPROVED** [review]

**Pull request merged**